### ISSUE TITLE
[DataGridPro] Horizontal scroll for grouping column

### ISSUE THREAD

**Issue by gkokoszkaFP** [mui/mui-x#5760]
### Order ID 💳

45646

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

I was trying to add new horizontal scroll for grouped column, if structure is very deep and we disable `resizible` prop in `groupingColDef`, highlited row should have more items but those are not visible.
![image](https://user-images.githubusercontent.com/91261184/184115580-02def3cc-29fd-4210-83d4-58c4cc558a30.png)

I have modified css a little bit to achive somehow workable solution but it is far from perfect. Here is output of this css modifications
![image](https://user-images.githubusercontent.com/91261184/184115993-0c4bdd9e-2cd9-425d-8cd2-c78c718b764f.png)

We have horizontal scroll but when virtualization kick's in it is moved down and it is no longer visible.
Is there any way to make scroll always visible on the bottom of the grouping container regardless of virtualization row buffer?
Also is there a way to separate one big horizontal scroll into two (one for grouped column, other for rest of data)?

### Your environment 🌎

<details>
  <summary>npx @mui/envinfo</summary>

```
  System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 AMD Ryzen 7 PRO 4750U with Radeon Graphics
    Memory: 3.04 GB / 31.37 GB
  Binaries:
    Node: 16.10.0 - C:\Program Files\nodejs\node.EXE
    npm: 7.24.0 - C:\Program Files\nodejs\npm.CMD
  Managers:
    pip3: 21.2.4 - C:\Python310\Scripts\pip3.EXE
  Utilities:
    Git: 2.35.2.
  Virtualization:
    Docker: 20.10.17 - C:\Program Files\Docker\Docker\resources\bin\docker.EXE
  IDEs:
    VSCode: 1.70.0 - C:\Users\gkokoszka\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
    Visual Studio: 17.2.32616.157 (Visual Studio Professional 2022)
  Languages:
    Java: 11.0.12
    Python: 3.10.1
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (104.0.1293.47)
    Internet Explorer: 11.0.19041.1566
```
</details>

**Comment by DanailH** [comment]
Hi, @gkokoszkaFP thanks for raising this!
It's an interesting problem. I'm not sure how to advise in this situation. @flaviendelangle might have some ideas.
Regarding the CSS selectors - they should work tho. It's possible that the virtualization adds some inline styles and they overwrite the once you add.

**Comment by flaviendelangle** [comment]
Interesting indeed

Maybe we could support horizontal scroll for the pinned row

**Comment by DanailH** [comment]
> Interesting indeed,
> 
> It would require supporting horizontal scroll on a single column, which is currently not possible at all

Yes, the DOM is not just not prepped to handle that. I'm quite curious how @gkokoszkaFP was able to make it work by adding CSS.

**Comment by gkokoszkaFP** [comment]
It was mainy done with those two lines of scss code
```
'& .MuiDataGrid-pinnedColumns': {
      width: `${groupingColumnWidth}px`,
      overflowX: 'scroll',
},
'& .MuiDataGrid-pinnedColumns .MuiDataGrid-cell': {
      maxWidth: 'inherit !important',
},
```

together with following styles for each grouped cell
```
export const GroupingCellTypography = styled(TypographyBase)({
  fontSize: '12px',
  lineHeight: '16px',
  textOverflow: 'ellipsis',
  width: 'inherit',
});

export const GroupingCellBox = styled(Box, {
  shouldForwardProp: prop => prop !== 'rowNodeDepth',
})<TableRowProps>(({ rowNodeDepth, theme }) => {
  return {
    paddingLeft: rowNodeDepth * CONSTANTS.GROUPING_COLUMN_DEPTH_SPACE,
    height: '100%',
    borderBottom: `1px solid ${theme.palette.secondary.dark}`,
  };
});
```

**Comment by DanailH** [comment]
ah, you have the grouped column also pinned? So it's about adding horizontal scroll on pinned columns?

**Comment by flaviendelangle** [comment]
If the column is pinned, then adding a scroll behavior to the pinned container is probably something we could consider doing.

**Comment by gkokoszkaFP** [comment]
Yes, sorry for misunderstanding, it is about scroll on pinned column that happens to be also grouped one containing grouping rows.

**Comment by DanailH** [comment]
I guess we can see if it makes sense to add it, what do you think @m4theushw?

**Comment by gkokoszkaFP** [comment]
@DanailH any idea when or if this feature would be shipped?

**Comment by DanailH** [comment]
@gkokoszkaFP we haven't prioritized it yet. I don't think it's something that is critical for us right now. I will add the `waiting for upvotes` label to see if others will express interest in having that feature but right now I can't say when we will focus on this.
